runtime/src/serde_snapshot/newer.rs

```diff
@@ -279,8 +306,18 @@ impl<'a> TypeContext<'a> for Context {
     where
         R: Read,
     {
-        let bank_fields = deserialize_from::<_, DeserializableVersionedBank>(&mut stream)?.into();
+        let mut bank_fields: BankFieldsToDeserialize =
+            deserialize_from::<_, DeserializableVersionedBank>(&mut stream)?.into();
         let accounts_db_fields = Self::deserialize_accounts_db_fields(stream)?;
+        // Process extra fields
+        let lamports_per_signature: u64 = match deserialize_from(stream) {
+            Err(err) if err.to_string() == "io error: unexpected end of file" => Ok(0),
+            Err(err) if err.to_string() == "io error: failed to fill whole buffer" => Ok(0),
```

Comment:
Matching on error string seems brittle to me. Can an error type be used instead?

Reply:
This is the same method used for the `AccountsDbFields`, see `default_on_eof` in `deserialize_utils`. Although I agree not sure if there's a better method.